pubsublite/operations.go

```diff
@@ -0,0 +1,152 @@
+// interface.
+type PublishTime time.Time
+
+func (p PublishTime) setRequest(req *pb.SeekSubscriptionRequest) {
+	req.Target = &pb.SeekSubscriptionRequest_TimeTarget{
+		TimeTarget: &pb.TimeTarget{
+			Time: &pb.TimeTarget_PublishTime{tspb.New(time.Time(p))},
+		},
+	}
+}
+
+// EventTime is a message event timestamp. It implements the SeekTarget
+// interface.
+type EventTime time.Time
+
+func (e EventTime) setRequest(req *pb.SeekSubscriptionRequest) {
+	req.Target = &pb.SeekSubscriptionRequest_TimeTarget{
+		TimeTarget: &pb.TimeTarget{
+			Time: &pb.TimeTarget_EventTime{tspb.New(time.Time(e))},
+		},
+	}
+}
+
+// SeekSubscriptionOption is reserved for future options.
+type SeekSubscriptionOption interface{}
+
+// OperationMetadata stores metadata for long-running operations.
+type OperationMetadata struct {
+	// The target of the operation. For example, targets of seeks are
+	// subscriptions, structured like:
+	// "projects/PROJECT_ID/locations/ZONE/subscriptions/SUBSCRIPTION_ID"
+	Target string
+
+	// The verb describing the kind of operation.
+	Verb string
+
+	// The time the operation was created.
+	CreateTime time.Time
+
+	// The time the operation finished running. Is zero if the operation has not
+	// completed.
+	EndTime time.Time
+}
+
+func protoToOperationMetadata(o *pb.OperationMetadata) (*OperationMetadata, error) {
+	if err := o.GetCreateTime().CheckValid(); err != nil {
+		return nil, err
+	}
+	metadata := &OperationMetadata{
+		Target:     o.Target,
+		Verb:       o.Verb,
+		CreateTime: o.GetCreateTime().AsTime(),
+	}
+	if o.GetEndTime() != nil {
+		if err := o.GetEndTime().CheckValid(); err != nil {
+			return nil, err
+		}
+		metadata.EndTime = o.GetEndTime().AsTime()
+	}
+	return metadata, nil
+}
+
+// SeekSubscriptionOperation manages a long-running seek operation from
+// AdminClient.SeekSubscription.
+type SeekSubscriptionOperation struct {
+	op *vkit.SeekSubscriptionOperation
+}
+
+// Name returns the path of the seek operation, in the format:
+// "projects/PROJECT_ID/locations/LOCATION/operations/OPERATION_ID".
+func (s *SeekSubscriptionOperation) Name() string {
+	return s.op.Name()
+}
+
+// Done returns whether the seek operation has completed.
+func (s *SeekSubscriptionOperation) Done() bool {
+	return s.op.Done()
+}
+
+// Metadata returns metadata associated with the seek operation. To get the
+// latest metadata, call this method after a successful call to Wait.
+func (s *SeekSubscriptionOperation) Metadata() (*OperationMetadata, error) {
+	m, err := s.op.Metadata()
+	if err != nil {
+		return nil, err
+	}
+	return protoToOperationMetadata(m)
+}
+
+// Wait polls until the seek operation is complete and returns one of the
+// following:
+//  - nil if the operation is complete and succeeded.
+//  - failure reason if the operation is complete and failed.
+//  - error if polling the operation status failed due to a non-retryable error.
+func (s *SeekSubscriptionOperation) Wait(ctx context.Context) error {
```

Comment:
I think this should return a response type, even if that type contains nothing today. I think `pubsub` has used this strategy in the past, @hongalex correct me if I am wrong. This way you can start returning info in the future without the need of a new method/breaking-change.

Reply:
Done. I guess it doesn't hurt, even if it's empty.